dev-tools/scripts/addVersion.py

```diff
@@ -44,10 +44,9 @@ def edit(buffer, match, line):
   print('done' if changed else 'uptodate')
 
 def add_constant(new_version, deprecate):
-  # TODO: Modify for new SolrVersion class, see SOLR-15845
-  filename = 'lucene/core/src/java/org/apache/lucene/util/Version.java'
+  filename = 'solr/core/src/java/org/apache/solr/util/SolrVersion.java'
```

Comment:
This is completely untested. Will probably get to it during the 9.0 release, but if anyone facies to review, go ahead.

Or if anyone knows of a better way for Solr to manage its own version than a SolrVersion class with enums, that could also be a way to go. Perhaps we could use code generation from gradle to insert the version somewhere and parse that statically. This huge version class seems a bit overkill imo. I also know there are various Java implementations of SemVer classes out there that could be used instead.

Reply:
Do we even have a need in Solr to know what prior versions there have been? Or do we only need to know our own version? Any persisting of version can happen as a `X.Y.Z` string, that can be parsed back and compared.

The Gradle build could write out a `solrVersion.properties` file into the jar file for consumption by a very simple SolrVersion class? Just throwing out thoughts here. Shoot them down please :)